eth/_utils/numeric.py

```diff
@@ -119,9 +119,8 @@ def integer_squareroot(value: int) -> int:
             )
         )
 
-    x = value
-    y = (x + 1) // 2
-    while y < x:
-        x = y
-        y = (x + value // x) // 2
-    return x
+    with decimal.localcontext() as ctx:
+        # Set precision to 128, since the largest square root of a
+        # 256-bit integer is a 128-bit integer.
+        ctx.prec = 128
+        return int(decimal.Decimal(value).sqrt())
```

Comment:
Looks cooool...!
I think it's better to note the "256-bit integer" assumption in the docstring.